### PR TITLE
[IMP] point_of_sale: numpad behaviour & unselect orderline

### PR DESCRIPTION
**Pull request by manv-afk**
-Select order line only when I've just added a product to cart
-Ensure that no order line is selected anymore when opening an order (from floorplan, table selector, ticket screen, payment screen)
-Adapt tours tests to match this new behavior and limit the use of 'inLeftSide' to avoid going uselessly back and forth to the product list in mobile

task-id: 4167563

Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/point_of_sale/static/src/app/components/order_tabs/order_tabs.js [hunk 1]
```diff
@@ -36,7 +36,7 @@ export class OrderTabs extends Component {
         if (previousOrderScreen?.name === "PaymentScreen") {
             props.orderUuid = order.uuid;
         }
-
+        this.pos.get_order().deselect_orderline();
```
Comment (davidmonnom) [inline addons/point_of_sale/static/src/app/components/order_tabs/order_tabs.js:39]: I'm wondering something, since no orderline needs to be selected when arriving on product_screen, can tyou maybe just put a condition in `showScreen` method like:

```
if (screen === "ProductScreen") {
    this.pos.get_order().deselect_orderline();
}
```

Like that you don't need to put it everywhere. Just at one place and you ensure that it pass everytime.

File: addons/point_of_sale/static/tests/tours/pos_combo_tour.js [hunk 2]
```diff
@@ -23,31 +24,38 @@ registry.category("web_tour.tours").add("ProductComboPriceTaxIncludedTour", {
...
             Dialog.confirm(),
-            ...ProductScreen.selectedOrderlineHas("Office Combo"),
-            ...ProductScreen.clickOrderline("Combo Product 3"),
-            ...ProductScreen.selectedOrderlineHas("Combo Product 3", "1.0", "13.43"),
-            ...ProductScreen.clickOrderline("Combo Product 5"),
-            ...ProductScreen.selectedOrderlineHas("Combo Product 5", "1.0", "18.67"),
-            ...ProductScreen.clickOrderline("Combo Product 8"),
-            ...ProductScreen.selectedOrderlineHas("Combo Product 8", "1.0", "30.00"),
-
+            inLeftSide([
```
Comment (davidmonnom) [inline addons/point_of_sale/static/tests/tours/pos_combo_tour.js:27]: Why this change ? How it was working before ?

**Review by davidmonnom — COMMENTED** [review]
I see a lot of changes about `inLeftSide` why do we need it now ? :)

**Comment by manv-afk** [comment]
> I see a lot of changes about `inLeftSide` why do we need it now ? :)

Before, different functions used in the tours like `ProductScreen.clickOrderline`,  `ProductScreen.selectedOrderlineHas` or `ProductScreen.clickNumpad` were all using `inLeftSide` (which is just clicking the cart on mobile and going back to the product screen). Now the behavior is that when we go back to the cart on mobile, the order line is not selected anymore.

So I tried to use as little `inLeftSide` as possible to avoid having to manually reselect my order line on mobile after each call to this function. To do that, I've wrapped my steps inside one call to `inLeftSide` when it's possible. It could also help to speed up the test because it's not going uselessly back and forth to the product screen in mobile.

File: addons/point_of_sale/static/tests/tours/pos_combo_tour.js (hunk 2 above)
Comment (manv-afk) [reply addons/point_of_sale/static/tests/tours/pos_combo_tour.js:27]: Each call to `selectedOrderlineHas` was calling `inLeftSide`, which unselect the order line on mobile (cause it goes to the cart then go back to product screen), so by wrapping every steps inside one `inLeftSide` it doesn't unselect the order line each time

File: addons/point_of_sale/static/src/app/components/order_tabs/order_tabs.js (hunk 1 above)
Comment (manv-afk) [reply addons/point_of_sale/static/src/app/components/order_tabs/order_tabs.js]: Yes it's better, I've made the changes, and I'm waiting for run bot to see if there is failing tests

**Comment by davidmonnom** [comment]
@robodoo r+
